features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/SessionReplay.kt

```diff
@@ -6,17 +6,26 @@
 
 package com.datadog.android.sessionreplay
 
+import androidx.annotation.VisibleForTesting
 import com.datadog.android.Datadog
+import com.datadog.android.api.InternalLogger
 import com.datadog.android.api.SdkCore
 import com.datadog.android.api.feature.Feature.Companion.SESSION_REPLAY_FEATURE_NAME
 import com.datadog.android.api.feature.FeatureSdkCore
 import com.datadog.android.sessionreplay.internal.SessionReplayFeature
+import java.lang.ref.WeakReference
 
 /**
  * An entry point to Datadog Session Replay feature.
  */
 object SessionReplay {
 
+    @VisibleForTesting internal var currentRegisteredCore: WeakReference<SdkCore>? = null
```

Comment:
The `SessionReplay` being a singleton object, it will stay in VM memory indefinitely, so indeed, having a standard reference to the core could have some memory leak.

Reply:
Actually I am just wondering should we rely on GC to release the reference here? shouldn't we have something to unregister it explicitly?

Reply:
We're querying the weakreference and checking if the core is active so we should be ok. If either of those checks fails we overwrite the variable